src/Umbraco.Web.UI.Client/src/packages/core/icon-registry/icon-picker-modal/icon-picker-modal.token.ts

```diff
@@ -1,6 +1,8 @@
 import { UmbModalToken } from '@umbraco-cms/backoffice/modal';
+import type { UmbIconDefinition } from '../types.js';
 
 export interface UmbIconPickerModalData {
+	filter?: (item: UmbIconDefinition) => boolean;
```

Comment:
@nielslyngsoe I noticed this is much similar to static filer picker:
https://github.com/umbraco/Umbraco-CMS/blob/cbde6af9c0a2c5028c427143e6a554f30ecbd776/src/Umbraco.Web.UI.Client/src/packages/static-file/components/input-static-file/input-static-file.element.ts#L79-L80

Also `filter` were in general used in old backoffice dialogs/modals, where it for static filer picker is called `pickableFilter`.

Reply:
Nvm, it seems the `filter` function from `UmbPickerModalData` filter the collection data as we want here, while `pickableFilter` decides which items can be picker, e.g. in static file picker.